include/swift/AST/Decl.h

```diff
@@ -6094,6 +6094,11 @@ class ConstructorDecl : public AbstractFunctionDecl {
     return getAttrs().hasAttribute<RequiredAttr>();
   }
 
+  /// Whether the constructor's result is discardable
+  bool isResultDiscardable() const {
```

Comment:
I think unless we're going to adopt this everywhere (in which case it belongs on AbstractFunctionDecl, probably), you might as well just inline this into the call site. We don't consider it uncommon to ask about attributes directly.

Reply:
i see :) .. ok .. i'll inline the attribute check on call site.